Source/WebCore/platform/graphics/Damage.h

```diff
@@ -97,68 +98,62 @@ class Damage {
         makeFull(m_size);
     }
 
-    ALWAYS_INLINE void add(const Region& region)
-    {
-        if (region.isEmpty() || !shouldAdd())
-            return;
-
-        for (const auto& rect : region.rects())
-            add(rect);
-    }
-
-    void add(const IntRect& rect)
+    bool add(const IntRect& rect)
     {
         if (rect.isEmpty() || !shouldAdd())
-            return;
+            return false;
 
         const auto rectsCount = m_rects.size();
         if (!rectsCount || rect.contains(m_minimumBoundingRectangle)) {
             m_rects.clear();
             m_rects.append(rect);
             m_minimumBoundingRectangle = rect;
-            return;
+            return true;
         }
 
         if (rectsCount == 1 && m_minimumBoundingRectangle.contains(rect))
-            return;
+            return false;
 
         m_minimumBoundingRectangle.unite(rect);
         if (m_mode == Mode::BoundingBox) {
             ASSERT(rectsCount == 1);
             m_rects[0] = m_minimumBoundingRectangle;
-            return;
+            return true;
         }
 
         if (m_shouldUnite) {
             unite(rect);
-            return;
+            return true;
         }
 
         if (rectsCount == m_gridSize.unclampedArea()) {
```

Comment:
In general comments would have helped here a bit to follow the logic -- you're just modifying existing code, I'm aware of that, just wanted to mention it.